site/src/api/api.ts

```diff
@@ -231,6 +231,34 @@ export const getTemplateVersionByName = async (
   return response.data
 }
 
+export type GetPreviousTemplateVersionByNameResponse =
+  | TypesGen.TemplateVersion
+  | undefined
```

Comment:
Can we make a type for this on the BE? Maybe in `codersdk/templateversions.go`?

Reply:
Great!